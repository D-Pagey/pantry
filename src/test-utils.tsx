import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

const AllTheProviders: FC = ({ children }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            }
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, options?: any) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
