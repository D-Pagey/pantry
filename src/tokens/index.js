export const CATEGORY_CARD_WIDTH = '120px';