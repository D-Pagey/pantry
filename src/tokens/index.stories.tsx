import React from 'react';
import { colours } from '.';

export default { title: 'Colours' };

export const all = () => (
    <div style={{ display: 'grid', gridTemplateColumns: '1fr 1fr 1fr 1fr 1fr', gridGap: '1rem' }}>
        {Object.entries(colours).map(([name, color]) => (
            <div key={name}>
                <p>{name}</p>
                <div style={{ width: 50, height: 50, backgroundColor: color }} />
            </div>
        ))}
    </div>
);
