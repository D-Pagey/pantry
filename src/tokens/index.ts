export const MOBILE_HEADER_HEIGHT = '80px';
export const HEADER_HEIGHT = '100px';
export const PAGE_WIDTH = '1100px';

export const CATEGORY_CARD_HEIGHT = '115px';
export const CATEGORY_CARD_WIDTH = '144px';

export const EXPIRING_SOON_DAYS = 3;

export const colours = {
    black: 'black',
    blue: 'rgb(66, 133, 244)',
    brown: 'brown',
    darkGreen10: 'rgba(52, 136, 65, 0.1)',
    darkGreen20: 'rgba(52, 136, 65, 0.2)',
    darkGreen40: 'rgba(52, 136, 65, 0.4)',
    darkGreen60: 'rgba(52, 136, 65, 0.6)',
    darkGreen100: '#348841',
    darkGrey: '#7C7C7C',
    grey: '#B2B2B2',
    lightGreen: '#31bb47',
    lightGrey: '#767676',
    veryLightGrey: '#E5E5E5',
    orange: '#FCBA3B',
    red: '#EB5757',
    white: 'white'
};

export const mediaQuery = {
    laptop: '(min-width: 1024px)',
    largeMobile: '(min-width: 375px)',
    maxPage: `(min-width: ${PAGE_WIDTH})`,
    tablet: '(min-width: 765px)'
};

export const zIndex = {
    dive: -200,
    default: 1,
    levitate: 2,
    rise: 200,
    sky: 400,
    top: 600
};
