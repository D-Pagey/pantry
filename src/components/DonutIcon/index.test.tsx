import React from 'react';
import { render } from '../../test-utils';
import { colours } from '../../tokens';
import { DonutIcon } from '.';

const props = {
    colour: colours.orange
};

describe('DonutIcon component', () => {
    it('should render', () => {
        const { container } = render(<DonutIcon {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
