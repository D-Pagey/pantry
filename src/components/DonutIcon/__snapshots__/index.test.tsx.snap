// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutIcon component should render 1`] = `
<svg
  fill="none"
  height="46"
  viewBox="0 0 46 46"
  width="46"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clip-rule="evenodd"
    d="M23 41C32.9411 41 41 32.9411 41 23C41 13.0589 32.9411 5 23 5C13.0589 5 5 13.0589 5 23C5 32.9411 13.0589 41 23 41ZM23 46C35.7025 46 46 35.7025 46 23C46 10.2975 35.7025 0 23 0C10.2975 0 0 10.2975 0 23C0 35.7025 10.2975 46 23 46Z"
    fill="#FCBA3B"
    fill-rule="evenodd"
  />
</svg>
`;
